aws/resource_aws_elasticache_cluster.go

```diff
@@ -204,18 +209,18 @@ func resourceAwsElasticacheCluster() *schema.Resource {
 				Elem:     &schema.Schema{Type: schema.TypeString},
 				Set:      schema.HashString,
 			},
-			// A single-element string list containing an Amazon Resource Name (ARN) that
-			// uniquely identifies a Redis RDB snapshot file stored in Amazon S3. The snapshot
-			// file will be used to populate the node group.
-			//
-			// See also:
-			// https://github.com/aws/aws-sdk-go/blob/4862a174f7fc92fb523fc39e68f00b87d91d2c3d/service/elasticache/api.go#L2079
 			"snapshot_arns": {
-				Type:     schema.TypeSet,
+				Type:     schema.TypeList,
```

Comment:
Doesn't look like the logic change was applied for this type change:

```
	snaps := d.Get("snapshot_arns").(*schema.Set).List()
	if len(snaps) > 0 {
		s := expandStringList(snaps)
		req.SnapshotArns = s
		log.Printf("[DEBUG] Restoring Redis cluster from S3 snapshot: %#v", s)
	}
```

```suggestion
				Type:     schema.TypeSet,
```

Reply:
Yeah, missed changing the handling of the values. We don't have any tests for restoring from S3.

Created #17085